File: 454-4sum-ii/454-4sum-ii.py
class Solution:
    def fourSumCount(self, nums1: List[int], nums2: List[int], nums3: List[int], nums4: List[int]) -> int:
        dic=dict()
        for i  in nums1:
            for j in nums2:
                dic[i+j]=dic.get(i+j,0)+1
        ct=0
        for k in nums3:
            for l in nums4:
                ct+=dic.get(-(k+l),0)
        
        return ct

File: 454-4sum-ii/454-4sum-ii.cpp
class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        map<int,int> mp;
        for(auto i:nums1)
            for(auto j:nums2)
                mp[i+j]++;
        int ct=0;
        for(auto i:nums3)
            for(auto j:nums4)
                ct+=mp[-(i+j)];
        return ct;
    }
};